lib/encoding.neon

```diff
@@ -0,0 +1,159 @@
+    VAR status: Mode := Mode.BYTE1
+    
+    LET source: Array<Number> := src.toArray()
+
+    WHILE i < src.size() DO
+        VAR char: Number := source[i]
+        CASE status
+            WHEN Mode.BYTE1 DO
+                encoded.append(Base64EncodeTable[bitwise.shiftRight32(char, 2)])
+                char1 := bitwise.and32(bitwise.shiftLeft32(char, 4), 0x30)
+                status := Mode.BYTE2
+            WHEN Mode.BYTE2 DO
+                encoded.append(Base64EncodeTable[bitwise.or32(char1, bitwise.shiftRight32(char, 4))])
+                char2 := bitwise.and32(bitwise.shiftLeft32(char, 2), 0x3c)
+                status := Mode.BYTE3
+            WHEN Mode.BYTE3 DO
+                encoded.append(Base64EncodeTable[bitwise.or32(char2, bitwise.shiftRight32(char, 6))])
+                encoded.append(Base64EncodeTable[bitwise.and32(char, 0x3F)])
+                status := Mode.BYTE1
+        END CASE
+        INC i
+    END WHILE
+    
+    CASE (status)
+        WHEN Mode.BYTE2 DO
+            encoded.append(Base64EncodeTable[char1])
+            encoded.append("==")
+        WHEN Mode.BYTE3 DO
+            encoded.append(Base64EncodeTable[char2])
+            encoded.append("=")
+    END CASE
+    RETURN encoded
+END FUNCTION
+
+%|
+ |  Function: base64Decode
+ |
+ |  Decodes the provided Base64 string data into binary data.
+ |
+ |  Parameters:
+ |      src - The source Base64 to decode.
+ |
+ |%
+FUNCTION base64Decode(src: String): Bytes
+    VAR i: Number := 0
+    VAR char1, char2, char3: Number := 0
+    VAR decoded: Array<Number> := []
+    VAR mode: Mode := Mode.BYTE1
+
+    WHILE i < src.length() DO
+        VAR char: Number := Base64DecodeTable[ord(src[i])]
+        IF bitwise.and32(char, 0x80) > 0 THEN
+            INC i
+            NEXT WHILE % Eat chars that are not part of the Base64 alphabet, include \r and \n.
+        END IF
+        IF char = 0x40 THEN
+            EXIT WHILE % We're at the end, so bail out.
+        END IF
+        
+        CASE mode
+            WHEN Mode.BYTE1 DO
+                char1 := char
+                mode := Mode.BYTE2
+            WHEN Mode.BYTE2 DO
+                char2 := char
+                decoded.append(bitwise.and32(bitwise.or32(bitwise.shiftLeft32(char1, 2), bitwise.shiftRight32(char2, 4)), 0xFF))
+                mode := Mode.BYTE3
+            WHEN Mode.BYTE3 DO
+                char3 := char
+                decoded.append(bitwise.and32(bitwise.or32(bitwise.shiftLeft32(char2, 4), bitwise.shiftRight32(char3, 2)), 0xFF))
+                mode := Mode.BYTE4
+            WHEN Mode.BYTE4 DO
+                decoded.append(bitwise.and32(bitwise.or32(bitwise.shiftLeft32(char3, 6), char), 0xFF))
+                mode := Mode.BYTE1
+        END CASE
+        INC i
+    END WHILE
+
+    RETURN decoded.toBytes()
+END FUNCTION
+
+FUNCTION stringToBytes(s: String): Bytes
+    RETURN s.toBytes()
+END FUNCTION
+
+BEGIN MAIN
+    print("Running Base64Encode tests...")
```

Comment:
Do not print out messages. That's done by the test framework after you move this to a separate test file.